File: Competive Programming/Python/ProjectEuler/Problem 7/Problem7.c
#include <stdio.h>
#include <stdbool.h>


bool is_prime(unsigned long long int n) {
    if(n==1)
        return false;
    if(n==2 || n==3)
        return true;
    for(int i=2; i<=n/2; ++i) {
        if(n%i == 0)
            return false;
    }
    return true;
}


int main() {
    int count = 1;
    unsigned long long int n = 0;

    while (count <= 10001) {
        ++n;
        if(is_prime(n))
            count++;
    }

    printf("%llu\n", n);
    return 0;
}

File: Competive Programming/Python/ProjectEuler/Problem 7/Problem7.go
package main


import "fmt"


func is_prime(n uint64) bool {
    var i uint64;
    if n == 1 {
        return false;
    }

    if n==2 || n==3 {
        return true;
    }

    for i = 2; i <= n/uint64(2); i += 1 {
        if n%i == 0 {
            return false;
        }
    }
    return true;
}


func main() {
    count := 1
    num := uint64(0)

    for ;count<=10001; {
        num += 1
        if is_prime(num) {
            count += 1
        }
    }

    fmt.Println(num);
}
